mesonbuild/wrap/wrap.py

```diff
@@ -643,11 +643,13 @@ def get_netrc_credentials(self, netloc: str) -> T.Optional[T.Tuple[str, str]]:
         if self.netrc is None or netloc not in self.netrc.hosts:
             return None
 
-        login, account, password = self.netrc.authenticators(netloc)
-        if account is not None:
-            login = account
+        info = self.netrc.authenticators(netloc)
+        if info is None:
+            return None
+
+        _, account, password = info
 
-        return login, password
+        return account, password
```

Comment:
You say we don't need to handle account being None "since account can never be None", but why? Particularly since the functionality logic here is "if account was present in the netrc file, prefer it instead of login".

You raise an interesting question in this commit, but the answer to it is, I think, that typing isn't everything, and that changing functional logic because typing says to is not necessarily correct, ***even if*** the change is correct according to typing.

... Although this change is NOT correct according to typing, if you take a close look at `typeshed/stdlib/netrc.pyi`.